packages/framework/src/Model/Order/Item/OrderItem.php

```diff
@@ -192,6 +212,42 @@ public function getPriceWithVat(): Money
         return $this->priceWithVat;
     }
 
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money|null
+     */
+    public function getTotalPriceWithoutVat(): ?Money
+    {
+        return $this->hasForcedTotalPrice() ? $this->totalPriceWithoutVat : null;
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function getTotalPriceWithVat(): Money
+    {
+        return $this->hasForcedTotalPrice() ? $this->totalPriceWithVat : $this->priceWithVat->multiply($this->quantity);
+    }
+
+    /**
+     * The total price property can be used when order item has prices that differ from current price calculation implementation.
+     * Otherwise it should be set to NULL (which means it will be calculated automatically).
+     *
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Price|null $totalPrice
+     */
+    public function setTotalPrice(?Price $totalPrice): void
+    {
+        $this->totalPriceWithVat = $totalPrice !== null ? $totalPrice->getPriceWithVat() : null;
```

Comment:
this method doesn't set prices that are in the argument... it's a mistake in my opinion. If the behavior is correct, then the function is wrongly named

Reply:
I don't get your point. This setter serves for the purpose of setting either both total prices at the same time, or unsetting them (saying the total price should be set to `null`). Seeing that `Price` is just a value object for holding two monetary values (a with VAT / without VAT price pair) I don't think it could be misleading...

Also, it's consistent with `Order::setTotalPrice()`.

Reply:
My apologiess, I read it as `... ? $this->getPriceWithVat() : null`

no problem in this place